src/main/java/com/sumologic/log4j/aggregation/BufferFlusherThread.java

```diff
@@ -84,6 +93,7 @@ private void flushAndSend()
       final Out body = aggregate(messages);
       sendOut(body);
       timeOfLastFlush = System.currentTimeMillis();
+      messageQueue.setError(false);
```

Comment:
there is no guarantee you captured the error event in this flush round. This is racy.